lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
Yes but AFAIK `concat` always creates a new buffer to copy the ones passed. I propose to avoid copying both buffers if the first one contains enough space to fit the second buffer and just copy the second buffer to the first one, hence 'append'.